src/solana/transaction.py

```diff
@@ -333,10 +335,14 @@ def verify_signatures(self) -> bool:
             return False
         return True
 
-    def serialize(self) -> bytes:
+    def serialize(self, verify_signatures: bool = True) -> bytes:
```

Comment:
Why is this change necessary?

Reply:
* Added verify signatures for transaction.serialize() since sometimes the frontend is the fee payer.
This uses the same functionality as [web3.js](https://github.com/solana-labs/solana/blob/363d8d1539e8cebc2d0921024d9edc35a38a4e34/web3.js/src/transaction.ts#L706)

^^ referenced in OP

so basically if the fee payer has not signed the txn, serialize() would fail if verify signatures is True. 

This is necessary for platforms that use the same design which is as follows: 
* backend creates txn and signs it (verify_signatures = False)
* send txn to frontend
* fee payer signs with (verifySignature = True) in web3.js

they even have requireAllSignature which hasn't been implemented in solana-py.